### ISSUE TITLE
Fails to build with ILP64 on Alpine Linux x86_64 since arpack 3.9.0

### ISSUE THREAD

**Issue by jirutka** [opencollab/arpack-ng#452]
## Expected behavior

Should build fine.

## Actual behavior

Fails to build.

## Where/how to reproduce the problem

- arpack-ng: 3.9.1
- OS: Alpine Linux Edge x86_64
- compiler: gcc (Alpine 13.2.1_git20240309)
- environment: FFLAGS=-fallow-argument-mismatch
- configure: `./configure --prefix=/usr --enable-static --enable-icb --with-blas=openblas64_ --with-lapack=openblas64_ LIBSUFFIX=64_ SYMBOLSUFFIX=_64 INTERFACE64=1`

## Steps to reproduce the problem

On Alpine Linux Edge x86_64:
```
apk add build-base openblas-dev autoconf automake gfortran libtool
./bootstrap
./configure \
	--build=$CBUILD \
	--host=$CHOST \
	--prefix=/usr \
	--enable-shared \
	--enable-static \
	--enable-icb \
	--with-blas=openblas64_ \
	--with-lapack=openblas64_ \
	LIBSUFFIX=64_ \
	SYMBOLSUFFIX=_64 \
	INTERFACE64=1
make
```

## Error message

```
snaup2.f:462:72:

  462 |          call scopy(kplusp, bounds, 1, workl(kplusp**2+2*kplusp+1), 1)
      |                                                                        1
Error: Syntax error in argument list at (1)
```

See [arpack-3.9.1-build.log](https://github.com/opencollab/arpack-ng/files/14816437/arpack-3.9.1-build.log) for full log.

## Traces
N/A

## Callstack

N/A

## Notes, remarks

Non-ILP64 builds fine. ILP64 builds fine for arpack-ng 3.8.0.

**Comment by fghoussen** [comment]
> Error: Syntax error in argument list at (1)

Is your BLAS / LAPACK  compiled with ILP64 support?

https://github.com/opencollab/arpack-ng/blob/51ce061ff7066ce712638bb058c805c40a326582/.github/workflows/jobs.yml#L156

**Comment by jirutka** [comment]
Yes, `--with-blas=openblas64_ --with-lapack=openblas64_` is from [openblas-ilp64](https://pkgs.alpinelinux.org/package/edge/community/x86_64/openblas-ilp64) which is OpenBLAS ILP64 variant (see [here](https://gitlab.alpinelinux.org/alpine/aports/-/blob/ace0432ca407b17dcd171e4cf57a33cfb5b51af4/community/openblas/APKBUILD#L101-106)).

**Comment by fghoussen** [comment]
Never tested with OpenBLAS ILP64, but only with MKL ILP64. Does OpenBLAS test suite runs OK with ILP64?

**Comment by fghoussen** [comment]
May be related: https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=967951

**Comment by fghoussen** [comment]
> ILP64 builds fine for arpack-ng 3.8.0.

With which version of `OpenBLAS`? Does it still work with this old specific version of `OpenBLAS`?
I had a quick look: I do not see any change that could break ILP64 at `arpack-ng` side.

**Comment by jirutka** [comment]
As I said, arpack-ng 3.8.0 builds fine with the same flags for ILP64 and the same OpenBLAS version, 3.9.0 fails to build.

**Comment by fghoussen** [comment]
As you use `--with-blas=openblas64_`, I guess (?) the `openblas` build/install you use would have produced `64_` suffixed openblas symbols and binaries (.a or .so file)? If so, you may try to (or should?) use `SYMBOLSUFFIX=64_` (underscore last) instead of `SYMBOLSUFFIX=_64` (underscore first)? Or maybe simply  `SYMBOLSUFFIX=_`? Or any other suffix according to the convention the openblas lib did use to export symbols in this specific ILP64 case?

How are `openblas` libraries named, and, how does they export symbols? (I have no idea - I do not use `openblas`)
In `openblas`, do you link with the expected ILP64 suffixed lib (static-archive or shared-object) referencing (using `SYMBOLSUFFIX`) the expected ILP64 symbols inside these libs? You may check that with `nm`: the ILP64 `openblas` lib **must** export the *same* symbols as the ones `arpack-ng` expect based on `SYMBOLSUFFIX` (check `FFLAGS` reported by `arpack-ng` configure log - summary at the end).

If `arpack-ng` reports:
```
--------------------------------------------------
Configuration summary for ARPACK-NG 3.9.1
--------------------------------------------------
Installation prefix : /usr
MPI enabled         : no (ICB provided )
ICB enabled         : yes
INTERFACE64         : 1
F77                 : x86_64-alpine-linux-musl-gfortran
FFLAGS              :  -fallow-argument-mismatch -fdefault-integer-8  -cpp -Dsaxpy=saxpy_64
```
Then the `openblas` lib **must** report (for `saxpy` and all other symbols like `scopy`):
```
>> nm /path/to/libopenblas64_.so | grep saxpy_64
0000000000000000 T saxpy_64
```
Is this the case at your side?
 
https://github.com/opencollab/arpack-ng/blob/51ce061ff7066ce712638bb058c805c40a326582/configure.ac#L288-L310   

AFAIK, the common BLAS/LAPACK libraries (netlib, openblas, ...) do not provide ISO_C bindings (huge work) that would enable the compiler to handle/detect types correctly (int or int64_t)... There is no way to automate/detect correct symbols to use: the only way is to rely on ugly  `SYMBOLSUFFIX` trick which means basically linking-by-hand: for this, you need to know for *sure* which symbol is exported how-and-why by which lib (in case you have different flavors of the same .so installed in the same install directory as it happens sometimes).

> arpack-ng 3.8.0 builds fine

Did the test suite run OK?
Sometimes these kind of errors may be silent at build time but will trigger failures (core dumps) at run time.

**Comment by fghoussen** [comment]
> I guess (?) the openblas build/install you use would have produced 64_ suffixed

Another test you may try is to install an ILP64 version of  `openblas` without any suffixing: this way there would be only one lib exporting always the same good-old expected-and-regular symbols (`scopy` symbol == ILP64 symbol) so you can drop `SYMBOLSUFFIX` at `arpack-ng` side.
